cli/azd/pkg/infra/azure_resource_manager_test.go

```diff
@@ -119,22 +137,36 @@ func TestGetDeploymentResourceOperationsSuccess(t *testing.T) {
 	mockContext := mocks.NewMockContext(context.Background())
 	scope := NewSubscriptionScope(*mockContext.Context, "eastus2", "SUBSCRIPTION_ID", "DEPLOYMENT_NAME")
 
-	mockContext.CommandRunner.When(func(args exec.RunArgs, command string) bool {
-		return strings.Contains(command, "az deployment operation sub list")
-	}).RespondFn(func(args exec.RunArgs) (exec.RunResult, error) {
+	mockContext.HttpClient.When(func(request *http.Request) bool {
+		return request.Method == http.MethodGet && strings.Contains(
+			request.URL.Path,
+			"/subscriptions/SUBSCRIPTION_ID/resourcegroups/resource-group-name/deployments/group-deployment-id/operations",
+		)
+	}).RespondFn(func(request *http.Request) (*http.Response, error) {
 		subCalls++
```

Comment:
FYI - We have a couple little test helper functions to make creating mock http responses a little easier.  `mocks.CreateHttpResponseWithBody(...)` and `mocks.CreateEmptyHttpResponse(...)`.  They both handle any JSON marshelling and creating the body reader/writer.  We are using these successfully in many other tests.

Reply:
uff, that would've been easier :) .  I had to type strings :)

I'll use it for the last API that I need to do.

And hopefully in the future we don't have to keep manually creating mocks :)